src/lambda/commands/uploadLambda.ts

```diff
@@ -444,52 +443,57 @@ async function uploadZipBuffer(
 
 export async function findApplicationJsonFile(startPath: vscode.Uri): Promise<vscode.Uri | undefined> {
     const isTemplateFile = /template.(yaml|yml|json)$/.test(startPath.fsPath)
-    const parentDir = isTemplateFile ? path.dirname(startPath.fsPath) : startPath.fsPath
+    const parentDir = isTemplateFile ? vscode.Uri.joinPath(startPath, '..') : startPath
 
     if (isCloud9()) {
-        const found = SystemUtilities.findFile(parentDir, '.application.json')
-        return found ? vscode.Uri.parse(found) : undefined
+        const found = await findFile(parentDir.fsPath, '.application.json')
```

Comment:
still wondering why `vscode.workspace.findFiles()` doesn't work. If it's because of cloud9, then we would want a wrapper similar to how this is implemented: https://github.com/aws/aws-toolkit-vscode/blob/00bdbbe9109eb5f4b78805a92b33b31497a99e85/src/shared/systemUtilities.ts#L54

Reply:
`findFiles` in general works for cloud9, but I think the real issue is that it we're looking for a dot file. I've tried different glob patterns to see if I can explicitly look for `.application.json` but none of the efforts have worked. The patterns work for vs code, but I'm unsure how to get them to work for cloud9.

Reply:
That link suggests using `RelativePattern`, which I'm already using.